MaxText/tests/gpt3_test.py

```diff
@@ -85,7 +85,7 @@ def setUp(self):
     }
     self.model_vars = init_random_model_vars(self.model, self.rng, self.example_batch)
 
-  @pytest.mark.tpu
+  @pytest.mark.tpu_only
   def test_logits_numerically(self):
```

Comment:
Yeap. You are right. This test is simply to avoid numerical discrepancy caused by any unexpected implementation changes in sharing module like attention or linear layers.

We don't care about the exact value of the model weight. A fixed random seed ensures a frozen model weights. And we do expect the same results with the same model weights as long as the implementation is not changed.  

The root cause might be relevant to a changing of random state, which might be affected by other test in parallel. I propose to double check random state in parallel tests first. Otherwise, we can switch to ckpt loading if there's hard conflict between random state and parallel testing.

<br/><br/>
> In an offline process you pass the same random model weights to get ground truth in a paxml implementation (how is this done, how do you ensure the same model weights are used?)

This was completed offline during the #293 see [numerical test](https://colab.sandbox.google.com/drive/1PuOiJTd9dGXheWzNpSYU_MEwpN-i6dyT?usp=sharing). Paxml environment is hard to setup and might introduce some conflicts.

<br/><br/>
> Won't this test fail if we pass different rng or jax changes its rng implementation since the model weights will change? 

Yeap. The goal is to verify the same logit outputs with fixed rng and  random weights while we may not care too much about the exact value of random weights. I think rng implementation might probably pretty stable in jax. 

<br/><br/>
> I think this correctness test makes more sense as an e2e test that loads a checkpoint from GCS.

You are right. It would be ideal to avoid dependency to random initialization. However, it might be hard to maintain for a long time since we need to change orbax ckpt each time
* we change the layer name
* change ckpt implementation
* some updates in orbax library

This happened in last mlperf 4.1 submission.

As for e2e test, there's a functional [e2e test](https://github.com/AI-Hypercomputer/maxtext/blob/main/end_to_end/tpu/test_gpt3.sh) for ckpt conversion and loading.